# 异常处理

`try-catch` 结构构成 C++ 异常处理，表示异常处理块中的捕获块，实现处理错误状态的代码与正常代码分开

```c++
try {
    // do something
} catch (const std::exception &e) {
    // do something
} catch (const std::exception&) {
    // do something
} catch (...) {
    // do something
}
```

```ad-danger
注意：异常处理不应替代程序中应有的数据检查和验证，异常处理带来的系统开销相当大
```

可捕获的类型不应是以下类型及其数组，引用和指针：
- 除 `void` 外的[[不完整类型]]
- 抽象类类型
- 右值引用

当我们不关心具体的异常值时，异常变量可省略（示例第二个 `catch` 块）

当我们使用 `...` 作为捕获变量时，表示该块可以捕获任意类型的异常

# 产生异常

#未完成 

`catch`中可直接使用 `throw` 重新抛出未捕获的异常

# 异常类

#未完成 

内存申请相关的异常为 `new` 头文件中的 `bad_alloc&` 类型

MFC 异常为 `CException*` 类型，并使用 `Delete()` 方法释放
